app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductActivity.java

```diff
@@ -275,21 +276,22 @@ private void setupViewPager( ViewPager viewPager )
 			adapterResult.addFragment( new NutritionInfoProductFragment(), menuTitles[3] );
 			if( PreferenceManager.getDefaultSharedPreferences( this ).getBoolean( "photoMode", false ) )
 			{
-				adapterResult.addFragment( new ProductPhotosFragment(), "Product Photos" );
+				adapterResult.addFragment( new ProductPhotosFragment(), getString(R.string.product_photos_title) );
 			}
 		}
 
 		if( BuildConfig.FLAVOR.equals( "obf" ) )
 		{
 			if( PreferenceManager.getDefaultSharedPreferences( this ).getBoolean( "photoMode", false ) )
 			{
-				adapterResult.addFragment( new ProductPhotosFragment(), "Product Photos" );
+				adapterResult.addFragment( new ProductPhotosFragment(), getString(R.string.product_photos_title) );
 			}
+			adapterResult.addFragment( new IngredientsAnalysisProductFragment(), getString(R.string.ingredient_analysis_title) );
```

Comment:
You must extract the the string in `menuTitles` string array. Do the same for product photos.

Reply:
okay

Reply:
It does not matter as you indexing . They wont propagate to all versions 😊

Reply:
Can you specify in which strings.xml file should I add? I've added items once in strings.xml, but i'm getting ArrayIndexOutOfBounds exception

Reply:
You can refer to this PR https://github.com/openfoodfacts/openfoodfacts-androidapp/commit/3b7a9084f44af668724e0a53d06cebdc93f31088

Reply:
I did in the same way but getting the exception. But when I added items in strings.xml(en) it is working fine (As the language code in my tested phone is en). So to get the string through menuTitles, I think we need to add items to every file in their respective languages. Is there any other way to solve this problem?